src/main/java/org/springframework/data/ldap/repository/support/SimplePagingLdapRepository.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.ldap.repository.support;
+
+import org.springframework.data.domain.*;
+import org.springframework.data.ldap.repository.PagingLdapRepository;
+import org.springframework.ldap.control.PagedResultsCookie;
+import org.springframework.ldap.control.PagedResultsDirContextProcessor;
+import org.springframework.ldap.core.ContextMapper;
+import org.springframework.ldap.core.DirContextOperations;
+import org.springframework.ldap.core.LdapOperations;
+import org.springframework.ldap.filter.Filter;
+import org.springframework.ldap.odm.annotations.Entry;
+import org.springframework.ldap.odm.core.ObjectDirectoryMapper;
+import org.springframework.ldap.query.LdapQuery;
+import org.springframework.ldap.support.LdapUtils;
+import org.springframework.util.Assert;
+
+import javax.naming.Name;
+import javax.naming.NamingException;
+import javax.naming.directory.SearchControls;
+import java.util.List;
+
+/**
+ * Paging repository implementation for LDAP.
+ *
+ * @author Houcheng Lin
+ */
+public class SimplePagingLdapRepository<T> extends SimpleLdapRepository<T> implements PagingLdapRepository<T> {
```

Comment:
We usually keep the whole base repository implementation in a single class. Could you move these methods to `SimpleLdapRepository`?